compiler/passes/scopeResolve.cpp

```diff
@@ -1043,6 +1043,12 @@ static void build_constructor(AggregateType* ct) {
       if (method->numFormals() == 2) {
         CallExpr* init = new CallExpr("initialize", gMethodToken, fn->_this);
         fn->insertAtTail(init);
+        // If a type has an initialize method, it's not Plain Old Data.
+        // The only time classes aren't Plain Old Data is when they have the
+        // pragma "no object" associated with them.
+        if (!isClass(ct) || ct->symbol->hasFlag(FLAG_NO_OBJECT)) {
+          ct->symbol->addFlag(FLAG_NOT_POD);
+        }
         break;
```

Comment:
It's actually to handle sync/single variables. I'll update the comment.
I'm not sure if classes can have user assignment overloads. I think that's a design decision we havn't made yet.

I agree about marking each type with FLAG_POD or FLAG_NOT_POD, but I think we have to make some decisions about classes (e.g.) to do that, and would like to put that off for another PR.

I'm adding the comment:

+        // Note: the use of FLAG_NO_OBJECT here is actually intended
+        // to capture sync and single variables. At least it indicates
+        // something is not a normal class instance.